### ISSUE TITLE
Appimage Installer Script won't ever update 

### ISSUE THREAD

**Issue by LeMoonStar** [zen-browser/desktop#2169]
### Captchas

- [X] I have read the instructions.
- [X] I have searched existing issues and avoided creating duplicates.
- [X] I am not filing an enhancement request.

### What happened?

When using the AppImage installer script, no update can be performed.

This is the case both, with `bash <(curl https://updates.zen-browser.app/appimage.sh)`  
and `bash <(curl https://updates.zen-browser.app/appimage.sh) twilight`.

Choosing the option `3) Check for Updates` (provided zsync is installed) always results in the message `(｡♥‿♥｡) Congrats! Zen Browser is up-to-date!`.
However, checking the version of the browser - It does not change.

<details><summary>Script Output</summary>
<p>

```
aaron@aaron-acralis-fedora:~/Downloads$ bash <(curl https://updates.zen-browser.app/appimage.sh)
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100 12497  100 12497    0     0  65126      0 --:--:-- --:--:-- --:--:-- 65429
[i] ╔════════════════════════════════════════════════════╗
[i] ║                                                    ║
[i] ║    (ﾉ◕ヮ◕)ﾉ*:･ﾟ✧  Zen Browser Installer            ║
[i] ║                                                    ║
[i] ║    CPU: AVX2 Supported (Optimized Version)         ║
[i] ║    Status: Zen Browser Installed                   ║
[i] ║    zsync: Installed (Needed for Updates)           ║
[i] ║                                                    ║
[i] ╚════════════════════════════════════════════════════╝
[i] 
[i] (★^O^★) What would you like to do?
[i]   1) Install
[i]   2) Uninstall
[i]   3) Check for Updates
[i]   0) Exit
Enter your choice (0-3): 3
[i] 
[w] Auto dedecting AVX2 support...
[i] Checking for updates...
[i] Checking for Update...
########################################################################################################################################################################### 100.0%
[i] Checking for Update successfully!
[i] (｡♥‿♥｡) Congrats! Zen Browser is up-to-date!
[i] 
[i] Thank you for using Zen Browser Installer!
aaron@aaron-acralis-fedora:~/Downloads$
```

</p>
</details> 

Digging a bit deeper reveals the procedure used by the script (Which can be found [here](https://github.com/zen-browser/updates-server/blob/18afa4eb5088f8d72aa00a17c9cadf0efbb6618e/appimage.sh)):
1. Download the zsync file.
2. Run `zsync -i ~/.local/share/AppImage/$file_base.AppImage -o ~/.local/share/AppImage/$file_base.AppImage "$zsync_file"` and capture it's output.
3. (try to) Check if the Output contains `verifying download...checksum matches OK` using `if echo "$update_output" | grep -q "verifying download...checksum matches OK"; then`
  - If so, display `(｡♥‿♥｡) Congrats! Zen Browser is up-to-date!`  
  - Otherwise
    1. Download the AppImage (not the zsync file!) into ` ~/.local/share/AppImage/$file_base.AppImage`
    2. "Process the AppImage" - extract Icons, .desktop file and so on.
    3. Display `(｡♥‿♥｡) Zen Browser updated to the latest!`
4. Delete the zsync file

We quickly notice:
- The output is always `(｡♥‿♥｡) Congrats! Zen Browser is up-to-date!` - so the update mechanism doesn't even kick in.
- The script only uses zsync to detect if there is an update, and then downloads the whole file itself. ZSync is made to do this itself. This indicates the developer of the script might have struggled to get zsync to work and created this workaround - then assumed it worked as they were already on the latest version.

So the source of the problem is likely to be the ZSync command.
Downloading the ZSync file and then executing the command as used in the script gives us the following output:

 ```
 aaron@aaron-acralis-fedora:~$ zsync -i ~/.local/share/AppImage/ZenTwilight.AppImage -o ~/.local/share/AppImage/ZenTwilight.AppImage ~/Downloads/ZenTwilight.AppImage.zsync 
Read /home/aaron/.local/share/AppImage/ZenTwilight.AppImage. Target 13.4% complete.      **********************************************************************************
Read /home/aaron/.local/share/AppImage/ZenTwilight.AppImage.part. Target 15.8% complete.      **********************************************************************************
URL 'zen-specific.AppImage' from the .zsync file is relative, but I don't know the referer URL (you probably downloaded the .zsync separately and gave it to me as a file). I need to know the referring URL (the URL of the .zsync) in order to locate the download. You can specify this with -u (or edit the URL line(s) in the .zsync file you have).
failed to retrieve from zen-specific.AppImage
Aborting, download available in /home/aaron/.local/share/AppImage/ZenTwilight.AppImage.part
verifying download...aaron@aaron-acralis-fedora:~$
```

This clearly is not correct. And even specifying the URL instead of downloading the file results in an error:

```
aaron@aaron-acralis-fedora:~$ zsync -i ~/.local/share/AppImage/ZenTwilight.AppImage -o ~/.local/share/AppImage/ZenTwilight.AppImage https://github.com/zen-browser/desktop/releases/download/twilight/zen-specific.AppImage.zsync
failed on url https://github.com/zen-browser/desktop/releases/download/twilight/zen-specific.AppImage.zsync
could not read control file from URL https://github.com/zen-browser/desktop/releases/download/twilight/zen-specific.AppImage.zsync
aaron@aaron-acralis-fedora:~$
```

Opening this URL in a browser results in a download, however, GitHub does this by redirecting the user. ZSync seems to be unable to follow this.
The grep command, meant to check if the ZSync output contains `verifying download...checksum matches OK` should not match, however, it seems the way the script checks this always matches - resulting in the illusion of there not being an update.

## TL;DR

The use of ZSync and Grep in this context makes the installer script unable to detect if there is an update.

### Reproducible?

- [X] I have checked that this issue cannot be reproduced on Mozilla Firefox.

### Version

1.0.1-a.*

### What platform are you seeing the problem on?

Linux

**Comment by LeMoonStar** [comment]
The script was first introduced in #1435 by @Muko-Tabi.
It has since been moved to [zen-browser/updates-server](https://github.com/zen-browser/updates-server)

**Comment by Samueru-sama** [comment]
Just in case my comments in the PR were not seen, it is worth mentioning that using the zsync file directly to check for updates isn't right as that file is meant for appimageupdatetool. 

A better method to check for updates that has very little dependencies is to parse the repo and store the download url, then when updates are checked we compare the new url with the old one to determine if there is an update avaiable, download and replace the old url file with the new one.

[TLDR](https://github.com/ivan-hc/AM/blob/main/programs/x86_64/zen-browser#L57-L87) and that update script has a check to use appimageupdatetool if it is installed.

**Comment by RayZ3R0** [comment]
Hello @LeMoonStar which linux distribution are you on? If you are on an Ubuntu based distro, you can try using `pacstall`

**Comment by LeMoonStar** [comment]
> Hello @LeMoonStar which linux distribution are you on? If you are on an Ubuntu based distro, you can try using `pacstall`

Hey! I am using Fedora Workstation.

However, this issue is not about solving the problem for me (*I know how to work around it*), much more as **a note that Zens own AppImage installer script is broken** and **a bit of investigation into why** it is broken.

**Comment by RayZ3R0** [comment]
Ah, okay. We'll look into this
